home/home_3.1/main.swift

```diff
@@ -0,0 +1,28 @@
+func calculateValue(x: Double, y: Double, z: Double) -> Double {
+    let product = x * y * z
+
+    if product != 0 {
+        return 1 / product
+    } else {
```

Comment:
Minor: sometimes it may be useful to linearise your code if possible especially if we already have return in one branch. something like:
```
if product != 0 {
    return 1 / product
}

let sum = x + y + z
if sum != 0 {
    return 1 / sum
}

return x + (y + 1) * (z - 1)
```
Of course it depends on the function specific and some potential scaling but especially for error handling with return such a way is more readable. Nested constructions like if for and so on may decrease readability for some cases